encoding/descaling.md

```diff
@@ -1,477 +1,332 @@
+You can spot one of those here,
+pointing towards a height of 720.
+When trying to run getnative,
+you ideally want to get a relatively bright frame
+with very clearly defined lineart
+and as little effects or filtering applied to it,
+like for example blur or glow.
+
+If you've got a graph without any clear spikes,
+it's usually a bad idea to try descaling that.
+It's likely that you're either working with native 1080p content,
+an already-descaled encode,
+or a source that has been too warped to properly descale.
+
+![](images/descale/descaled_fgo_op_encoded_frame.png)
+
+![Running getnative on frame of a already-rescaled encode](images/descale/descaled_fgo_op_encoded_graph.png)
+
+Also of note is that you should be careful
+to not pick a frame with credits to run getscaler on.
+Credits are usually added at the very end,
+after the clip has already been upscaled.
+Those will need to be handled separately.
+
+![](images/descale/descaled_yaiba_17_credits_frame.png)
+
+![Credits and other native 1080p elements mess with the graph](images/descale/descaled_yaiba_17_credits_graph.png)
+
+And last but not least,
+we have letterboxing.
+Letterboxing is what we call the black bars surrounding the video.
+As an added bonus,
+this example is also from a blu-ray release
+that was brutally post-processed,
+making it impossible to descale.
+
+![](images/descale/descale_paniponidash_op2_letterbox&qtec_frame.png)
+
+![A murder scene](images/descale/descale_paniponidash_op2_letterbox&qtec_graph.png)
 
 [getnative]: https://github.com/Infiziert90/getnative
-[descale]: https://github.com/BluBb-mADe/vapoursynth-descale
 
 
-## Finding out the native resolution
+# Filterchains
 
-To figure out what
-the native resolution of an anime is,
-first you need a good frame to test.
-Ideally,
-you'll want a bright frame
-with as little blur as possible of high quality
-(Blu-ray or very good webstreams).
-It also helps to not have
-too many post-processed elements in the picture.
-Whilst it is most definitely possible
-to get pretty good results with "bad" frames,
-it's generally better to use
-good frames whenever possible.
+Once you've figured out the native resolution,
+you can move on to the actual descaling.
+There's two common ways to do this.
+The first one is by using [descale][].
+The other is by using `invks` from [fmtc][].
+For convenience's sake,
+we'll be keeping it to just `descale` for this guide.
 
-Here are some examples of "bad" frames.
+There are a couple very common kernels
+used in anime production:
+* Bilinear
+* Bicubic b=1/3, c=1/3 (Mitchell-Netravali)
+* Bicubic b=0, c=0.5 (Catmull-Rom)
 
-![Manaria Friends — 01 (frame 1)](images/descale_manaria01.png)
+Occasionally, although rare,
+you may also run into the following kernels:
+* Lanczos taps=3
+* Lanczos taps=4
+* Spline16
+* Spline36
+* Bicubic b=1, c=0 (B-Spline)
+* Bicubic b=0, c=0 (Hermite)
+* Bicubic b=0, c=0 (Hermite)
+* Bicubic b=0.3782, c=0.3109 (Robidoux)
```

Comment:
I've never heard of Robidoux, but I'm not an encoder. Are these actually used?

Reply:
They've ended up being the best results for some stuff I've come across.